File: app/com/models/Tournement.java
package com.models;

import com.fasterxml.jackson.annotation.JsonIgnore;

import java.io.Serializable;
import javax.persistence.*;
import java.util.List;


/**
 * The persistent class for the tournement database table.
 * 
 */
@Entity
@NamedQuery(name="Tournement.findAll", query="SELECT t FROM Tournement t")
public class Tournement implements Serializable {
	private static final long serialVersionUID = 1L;

	@Id
	@Column(name="tournement_id")
	private int tournementId;

	@Column(name="tournement_date_end")
	private String tournementDateEnd;

	@Column(name="tournement_date_start")
	private String tournementDateStart;

	//bi-directional many-to-one association to Footmatch
    @JsonIgnore
	@OneToMany(mappedBy="tournement")
	private List<Footmatch> footmatches;

	//bi-directional many-to-many association to User
    @JsonIgnore
	@ManyToMany(mappedBy="tournements")
	private List<User> users;

	//bi-directional many-to-one association to UserTournement
	@OneToMany(mappedBy="tournement")
	private List<UserTournement> userTournements;

	public Tournement() {
	}

	public int getTournementId() {
		return this.tournementId;
	}

	public void setTournementId(int tournementId) {
		this.tournementId = tournementId;
	}

	public String getTournementDateEnd() {
		return this.tournementDateEnd;
	}

	public void setTournementDateEnd(String tournementDateEnd) {
		this.tournementDateEnd = tournementDateEnd;
	}

	public String getTournementDateStart() {
		return this.tournementDateStart;
	}

	public void setTournementDateStart(String tournementDateStart) {
		this.tournementDateStart = tournementDateStart;
	}

	public List<Footmatch> getFootmatches() {
		return this.footmatches;
	}

	public void setFootmatches(List<Footmatch> footmatches) {
		this.footmatches = footmatches;
	}

	public Footmatch addFootmatch(Footmatch footmatch) {
		getFootmatches().add(footmatch);
		footmatch.setTournement(this);

		return footmatch;
	}

	public Footmatch removeFootmatch(Footmatch footmatch) {
		getFootmatches().remove(footmatch);
		footmatch.setTournement(null);

		return footmatch;
	}

	public List<User> getUsers() {
		return this.users;
	}

	public void setUsers(List<User> users) {
		this.users = users;
	}

	public List<UserTournement> getUserTournements() {
		return this.userTournements;
	}

	public void setUserTournements(List<UserTournement> userTournements) {
		this.userTournements = userTournements;
	}

	public UserTournement addUserTournement(UserTournement userTournement) {
		getUserTournements().add(userTournement);
		userTournement.setTournement(this);

		return userTournement;
	}

	public UserTournement removeUserTournement(UserTournement userTournement) {
		getUserTournements().remove(userTournement);
		userTournement.setTournement(null);

		return userTournement;
	}

}

File: ui/src/app/models/tournement.ts
import {Column, Entity, ManyToOne, OneToMany} from 'entitype';

import { Footmatch } from './footmatch';
import { UserTournement } from './user-tournement';
import {Club} from "./club";

@Entity('tournement')
export class Tournement {

  @Column({ columnName: `tournement_id`, type: `int(11)`, nullable: false, generated: true, primaryKey: true, default: null })
  _tournementId: number;

  @Column({ columnName: `tournement_date_start`, type: `varchar(45)`, default: null })
  _tournementDateStart?: string;

  @Column({ columnName: `tournement_date_end`, type: `varchar(45)`, default: null })
  _tournementDateEnd?: string;

  @Column({ columnName: `club_id`, type: `int(11)`, nullable: false, default: null, index: true })
  clubId: number;

  @OneToMany(type => Footmatch, x => x.tournementId)
  _footmatches: Footmatch[];

  @ManyToOne(type => Tournement, x => x.clubId)
  club: Club;

  @OneToMany(type => UserTournement, x => x.tournementId)
  _userTournements: UserTournement[];


  public get tournementId(): number {
    return this._tournementId;
  }

  public set tournementId(value: number) {
    this._tournementId = value;
  }

  public get tournementDateStart(): string {
    return this._tournementDateStart;
  }

  public set tournementDateStart(value: string) {
    this._tournementDateStart = value;
  }

  public get tournementDateEnd(): string {
    return this._tournementDateEnd;
  }

  public set tournementDateEnd(value: string) {
    this._tournementDateEnd = value;
  }

  public get footmatches(): Footmatch[] {
    return this._footmatches;
  }

  public set footmatches(value: Footmatch[]) {
    this._footmatches = value;
  }

  public get userTournements(): UserTournement[] {
    return this._userTournements;
  }

  public set userTournements(value: UserTournement[]) {
    this._userTournements = value;
  }
}
